From 9cd159cff01621ed8524f8607626724cfd70aa26 Mon Sep 17 00:00:00 2001
From: cnauroth <cnauroth@apache.org>
Date: Wed, 29 Oct 2014 11:18:44 -0700
Subject: [PATCH 669/791] HADOOP-11068. Match hadoop.auth cookie format to
 jetty output. Contributed by Gregory Chanan.

(cherry picked from commit 5c900b522e70b512498ab7689f56eb1c5e4b963f)
---
 .../server/AuthenticationFilter.java               |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/hadoop-common-project/hadoop-auth/src/main/java/org/apache/hadoop/security/authentication/server/AuthenticationFilter.java b/hadoop-common-project/hadoop-auth/src/main/java/org/apache/hadoop/security/authentication/server/AuthenticationFilter.java
index f5a167b..0ac352b 100644
--- a/hadoop-common-project/hadoop-auth/src/main/java/org/apache/hadoop/security/authentication/server/AuthenticationFilter.java
+++ b/hadoop-common-project/hadoop-auth/src/main/java/org/apache/hadoop/security/authentication/server/AuthenticationFilter.java
@@ -609,9 +609,8 @@ public static void createAuthCookie(HttpServletResponse resp, String token,
     StringBuilder sb = new StringBuilder(AuthenticatedURL.AUTH_COOKIE)
                            .append("=");
     if (token != null && token.length() > 0) {
-      sb.append(token);
+      sb.append("\"").append(token).append("\"");
     }
-    sb.append("; Version=1");
 
     if (path != null) {
       sb.append("; Path=").append(path);
-- 
1.7.9.5

